src/CodeGen_D3D12Compute_Dev.cpp

```diff
@@ -576,7 +576,7 @@ void CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_C::visit(const Select *op) {
 
 void CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_C::visit(const Allocate *op) {
 
-    if (op->name == "__shared") {
```

Comment:
You'd make a local variable here `ScopedBinding<> bind_if(op->memory_type == MemoryType::GPUShared, shared_allocations, op->name);`